src/npc.cpp

```diff
@@ -234,16 +234,16 @@ void Npc::goToFollowCreature()
 	updateFollowCreaturePath(fpp);
 }
 
-void Npc::onCreatureAppear(Creature* creature, bool isLogin)
+void Npc::onCreatureAppear(std::shared_ptr<Creature> creature, bool isLogin)
 {
 	Creature::onCreatureAppear(creature, isLogin);
 
-	if (creature == this) {
+	if (creature.get() == this) {
```

Comment:
We’ve got some legacy code here, hehe!
```suggestion
	if (creature == getNpc()) {
```